transport-native-epoll/src/main/java/io/netty/channel/epoll/DomainSocketAddress.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel.epoll;
+
+import java.io.File;
+import java.net.SocketAddress;
+
+public final class DomainSocketAddress extends SocketAddress {
+    private final String socketPath;
+
+    public DomainSocketAddress(String socketPath) {
+        if (socketPath == null) {
+            throw new NullPointerException("socketPath");
+        }
+        this.socketPath = socketPath;
+    }
+
+    public DomainSocketAddress(File file) {
+        this(file.getAbsolutePath());
```

Comment:
The thing is, getAbsolutePath() never guarantees that the returned path string is unique.  You'll have to use getCanonicalPath() to get it, but it's expensive and it throws an exception.  So, unless there's a good reason, I would use getPath() for simplicity here.